src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -260,64 +263,56 @@ class GoogleExtensions(
   /**
     * Evaluate a future for each pet in parallel.
     */
-  private def forAllPets[T](userId: WorkbenchUserId)(f: PetServiceAccount => Future[T]): Future[Seq[T]] =
+  private def forAllPets[T](userId: WorkbenchUserId, samRequestContext: SamRequestContext)(f: PetServiceAccount => Future[T]): Future[Seq[T]] =
     for {
-      pets <- directoryDAO.getAllPetServiceAccountsForUser(userId).unsafeToFuture()
+      pets <- directoryDAO.getAllPetServiceAccountsForUser(userId, samRequestContext).unsafeToFuture()
       a <- Future.traverse(pets) { pet =>
         f(pet)
       }
     } yield a
 
-  override def onUserEnable(user: WorkbenchUser): Future[Unit] =
+  override def onUserEnable(user: WorkbenchUser, samRequestContext: SamRequestContext): Future[Unit] =
     for {
       _ <- withProxyEmail(user.id) { proxyEmail =>
         googleDirectoryDAO.addMemberToGroup(proxyEmail, WorkbenchEmail(user.email.value))
       }
-      _ <- forAllPets(user.id) { enablePetServiceAccount }
+      _ <- forAllPets(user.id, samRequestContext)({ (petServiceAccount: PetServiceAccount) => enablePetServiceAccount(petServiceAccount, samRequestContext) })
     } yield ()
 
-  override def onUserDisable(user: WorkbenchUser): Future[Unit] =
+  override def onUserDisable(user: WorkbenchUser, samRequestContext: SamRequestContext): Future[Unit] =
     for {
-      _ <- forAllPets(user.id) { disablePetServiceAccount }
+      _ <- forAllPets(user.id, samRequestContext)({ (petServiceAccount: PetServiceAccount) => disablePetServiceAccount(petServiceAccount, samRequestContext) })
       _ <- withProxyEmail(user.id) { proxyEmail =>
         googleDirectoryDAO.removeMemberFromGroup(proxyEmail, WorkbenchEmail(user.email.value))
       }
     } yield ()
 
-  override def onUserDelete(userId: WorkbenchUserId): Future[Unit] =
+  override def onUserDelete(userId: WorkbenchUserId, samRequestContext: SamRequestContext): Future[Unit] =
     for {
-      _ <- forAllPets(userId)(removePetServiceAccount)
+      _ <- forAllPets(userId, samRequestContext)((petServiceAccount: PetServiceAccount) => removePetServiceAccount(petServiceAccount, samRequestContext))
       _ <- withProxyEmail(userId) { googleDirectoryDAO.deleteGroup }
     } yield ()
 
   override def onGroupDelete(groupEmail: WorkbenchEmail): Future[Unit] =
     googleDirectoryDAO.deleteGroup(groupEmail)
 
-  @deprecated("Use new two-argument version of this function", "Sam Phase 3")
-  def createUserPetServiceAccount(user: WorkbenchUser): Future[PetServiceAccount] =
-    createUserPetServiceAccount(user, petServiceAccountConfig.googleProject).unsafeToFuture()
-
-  @deprecated("Use new two-argument version of this function", "Sam Phase 3")
-  def deleteUserPetServiceAccount(userId: WorkbenchUserId): Future[Boolean] =
-    deleteUserPetServiceAccount(userId, petServiceAccountConfig.googleProject).unsafeToFuture() //TODO: shall we delete these deprecated methods
```

Comment:
verified that they aren't used in the code elsewhere ; could remove this to be handled in a separate PR